runtime/server/auth/jwt.go

```diff
@@ -0,0 +1,223 @@
+package auth
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/MicahParks/keyfunc"
+	"github.com/golang-jwt/jwt/v4"
+	"go.uber.org/zap"
+	"gopkg.in/square/go-jose.v2"
+)
+
+// Issuer creates JWTs with claims for an Audience.
+// The Issuer is used by the admin server to create JWTs for the runtimes it manages based on a user's control-plane permissions.
+type Issuer struct {
+	issuerURL  string
+	signingKey jose.JSONWebKey
+	publicJWKS []byte
+}
+
+// NewIssuer creates an issuer from a JWKS. The JWKS must contain private keys.
+// The key identified by signingKeyID will be used to sign new JWTs.
+func NewIssuer(issuerURL, signingKeyID string, jwksJSON []byte) (*Issuer, error) {
```

Comment:
admin code for using this logic is still pending I think

Reply:
Yes, I thought I would do that in a separate PR. Although it might be easy to do, let me take a look

Reply:
was just pointing out, can be in another PR.